src/main/java/com/cleanroommc/groovyscript/compat/mods/prodigytech/Solderer.java

```diff
@@ -0,0 +1,160 @@
+
+    @MethodDescription(example = @Example("item('minecraft:iron_ingot')"))
+    public boolean removeByAdditive(IIngredient additive) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> r.requiresAdditive() && additive.test(r.getAdditive()))
+                .findFirst().orElse(null);
+        return remove(recipe);
+    }
+
+    @MethodDescription(example = @Example("item('prodigytech:circuit_refined')"))
+    public boolean removeByOutput(IIngredient output) {
+        SoldererManager.SoldererRecipe recipe = SoldererManager.RECIPES.stream()
+                .filter(r -> output.test(r.getOutput()))
+                .findFirst().orElse(null);
+        return remove(recipe);
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeAll() {
+        SoldererManager.RECIPES.forEach(this::addBackup);
+        SoldererManager.removeAll();
+    }
+
+    @MethodDescription(priority = 2000, example = @Example(commented = true))
+    public void removeWithoutAdditive() {
+        SoldererManager.RECIPES.removeIf(r -> {
+            if (r.requiresAdditive()) return false;
+            addBackup(r);
+            return true;
+        });
+    }
+
+    @MethodDescription(type = MethodDescription.Type.QUERY)
+    public SimpleObjectStream<SoldererManager.SoldererRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(SoldererManager.RECIPES)
+                .setRemover(this::remove);
+    }
+
+    @Property(property = "input", valid = @Comp("1"))
+    @Property(property = "output", valid = @Comp("1"))
+    public static class RecipeBuilder extends AbstractRecipeBuilder<SoldererManager.SoldererRecipe> {
+
+        @Property(valid = @Comp(value = "1", type = Comp.Type.GTE))
+        private int time = -1;
+
+        @Property(valid = @Comp(value = "1", type = Comp.Type.GTE))
+        private int gold;
+
+        @Property(valid = @Comp("1"))
+        private IIngredient pattern;
+
+        @RecipeBuilderMethodDescription
+        public Solderer.RecipeBuilder time(int time) {
+            this.time = time;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public Solderer.RecipeBuilder gold(int gold) {
+            this.gold = gold;
+            return this;
+        }
+
+        @RecipeBuilderMethodDescription
+        public Solderer.RecipeBuilder pattern(IIngredient pattern) {
+            this.pattern = pattern;
+            return this;
+        }
+
+        @Override
+        public String getErrorMsg() {
+            return "Error adding ProdigyTech Solderer Recipe";
+        }
+
+        @Override
+        public void validate(GroovyLog.Msg msg) {
+            validateItems(msg, 0, 1, 1, 1);
+            validateFluids(msg);
+            msg.add(gold <= 0, "gold must be greater than or equal to 1, yet it was {}", gold);
+            msg.add(pattern == null, "pattern cannot be empty");

```

Comment:
should check if its empty instead of null, since `IIngredient.EMPTY` exists.
```suggestion
            msg.add(IngredientHelper.isEmpty(pattern), "pattern must not be empty");
```